### DISCUSSION TITLE
A Kubernetes resource with a malformed image reference (`busybox:la!e$t`) crashes the whole scan

### DISCUSSION THREAD

**Discussion by asekretenko** [aquasecurity/trivy#5891] [Bugs]
### Description

A K8s deployment containing a malformed image reference (`busybox:la!e$t`) will crash `./trivy kubernetes -A all --report all` early, before it manages to scan any image.

Known affected versions:
- v0.45.1
- current `main` (47b6c2817a1791b6892868b6dedaf9b99a7a8e9a)

### Desired Behavior

- **Everything else is scanned.**

- [debatable] The exit code is 0 and the parse error is logged as a warning. I would argue that skipping such an image is safe, because it cannot be pulled. However, I'm not quite sure whether a combination of non-conforming container runtime/docker registry could still pull such an image.

### Actual Behavior

The scan fails early logging an error message (see Debug Output).

From what I can tell, for such an image [GetServerFromImageRef()](https://github.com/aquasecurity/trivy-kubernetes/blob/aaa27a0860784354c87b61b7273b9c00740e824f/pkg/k8s/docker/config.go#L118) simply returns an error,
which is propagated up the stack without any special handling, failing the whole of [clusterRun()](https://github.com/aquasecurity/trivy/blob/47b6c2817a1791b6892868b6dedaf9b99a7a8e9a/pkg/k8s/commands/cluster.go#L41).

### Reproduction Steps
1. Create a file `d.yaml` with the following contents:
```bash
apiVersion: apps/v1
kind: Deployment
metadata:
  name: invalid-image-name
  labels:
    app: invalid-image-name
spec:
  replicas: 1
  selector:
    matchLabels:
      app: invalid-image-name
  template:
    metadata:
      labels:
        app: invalid-image-name
    spec:
      containers:
      - name: busybox
        image: busybox:la!e$t
        args:
        - sleep
        - "86400"
```
2. Run `kubectl apply -f d.yaml`
3. Run `./trivy kubernetes -A all --report all`

### Target

Kubernetes

### Scanner

None

### Output Format

Table

### Mode

Standalone

### Debug Output

```bash
2024-01-08T12:30:19.592+0100	DEBUG	Severities: ["UNKNOWN" "LOW" "MEDIUM" "HIGH" "CRITICAL"]
2024-01-08T12:30:19.593+0100	DEBUG	Ignore statuses	{"statuses": null}
2024-01-08T12:30:20.234+0100	FATAL	get k8s artifacts error:
    github.com/aquasecurity/trivy/pkg/k8s/commands.namespaceRun
        <...>/d/trivy/pkg/k8s/commands/namespace.go:28
  - could not parse reference: busybox:la!e$t
```

### Operating System

macOS/Linux

### Version

```bash
Version: dev
```

### Checklist

- [X] Run `trivy image --reset`
- [X] Read [the troubleshooting](https://aquasecurity.github.io/trivy/latest/docs/references/troubleshooting/)

**Comment by chen-keinan** [comment]
Converted to issue #5895
